CRM/Contribute/BAO/Contribution.php

```diff
@@ -5030,4 +5030,29 @@ private static function getActivitySourceAndTarget($activityId): array {
     return [$sourceContactId, $targetContactId];
   }
 
+  /**
+   * Get the unit label with the plural option
+   *
+   * @param string $unit
+   * @return string
+   */
+  public static function getUnitLabelWithPlural($unit) {
+    switch ($unit) {
+      case 'day':
+        return ts('day(s)');
+
+      case 'week':
+        return ts('week(s)');
+
+      case 'month':
+        return ts('month(s)');
+
+      case 'year':
+        return ts('year(s)');
+
+      default:
+        throw new CRM_Core_Exception('Unknown unit: ' . $unit);
```

Comment:
It's worth noting that it's currently possible to edit the "Recurring Frequency Units" options from the "Administer" -> "Option Groups" area of the CiviCRM UI.

I can't think of any good reasons for adding or removing frequency units in this way, but there might be - if so it might be better to fallback to the exiting method of appending '(s)' if an unrecognised string is found, rather than throwing the exception.

Alternatively, maybe the "Recurring Frequency Units" option_group should be set to is_locked so that new options cannot be added. (and appreciate that's a seperate task, but maybe something to consider in tandem).

Reply:
@braders see comments of [this issue](https://lab.civicrm.org/dev/translation/-/issues/73): I would have happily done so: `return ts($unit . '(s)');` but we cannot put a variable in translation: https://docs.civicrm.org/dev/en/latest/translation/#best-practices.

Furthermore, the intervention is consistent with [this](https://github.com/civicrm/civicrm-core/pull/20308/files#diff-a66dfa276392f7ed78d3855b4d3694fa9463bbe7054610865112911e6bfabad3R113). See also [this comment](https://github.com/civicrm/civicrm-core/pull/20308#issuecomment-886969176).